sklearn/metrics/_dist_metrics.pxd

```diff
@@ -1,14 +1,18 @@
 #!python
-#cython: boundscheck=False
-#cython: wraparound=False
-#cython: cdivision=True
+# cython: annotate=False
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: profile=False
+# cython: linetrace=False
+# cython: initializedcheck=False
+# cython: binding=False
```

Comment:
That's right (I maily got them setup to be able to profile or `cythonize` a few file more easily).

> Does the last one have an impact ? I recall trying it once but it did not change anything. Maybe it does here, it would be interesting to check

Are you referring to `binding` or `initializedcheck`?

To clarify the intend:
 - I indicated the value of `binding` to preserve the implementation behavior because [its default value will change from False to True in Cython 3](https://cython.readthedocs.io/en/latest/src/userguide/source_files_and_compilation.html#compiler-directives). I don't really know what the best to set for this directive.
 - I specified `initializedcheck=False` to disable initialization checks of memoryviews because we don't need them and because they can be costly (see https://github.com/scikit-learn/scikit-learn/pull/20323#discussion_r657317558 for instance).

Reply:
I am waiting for your advice on handling `binding` before applying this suggestion and others regarding cython directives.

Reply:
Reading [the mention in the migration notes from Cython 0.29 to Cython 3.](https://cython.readthedocs.io/en/latest/src/userguide/migrating_to_cy30.html#binding-functions), it looks like it has no impact here. I am proceeding by removing the `binding` directive.